android/src/main/kotlin/com/onegini/mobile/sdk/flutter/PigeonInterface.kt

```diff
@@ -200,42 +198,47 @@ open class PigeonInterface : UserClientApi, ResourceMethodApi {
     deregisterUserUseCase(profileId, callback)
   }
 
-  override fun getRegisteredAuthenticators(profileId: String, callback: (Result<List<OWAuthenticator>>) -> Unit) {
-    callback(getRegisteredAuthenticatorsUseCase(profileId))
+  override fun getAuthenticatedUserProfile(callback: (Result<OWUserProfile>) -> Unit) {
+    callback(getAuthenticatedUserProfileUseCase())
   }
 
-  override fun getAllAuthenticators(profileId: String, callback: (Result<List<OWAuthenticator>>) -> Unit) {
-    callback(getAllAuthenticatorsUseCase(profileId))
+  override fun authenticateUser(
+    profileId: String,
+    authenticatorType: OWAuthenticatorType,
+    callback: (Result<OWRegistrationResponse>) -> Unit
+  ) {
+    authenticateUserUseCase(profileId, authenticatorType, callback)
   }
 
-  override fun getAuthenticatedUserProfile(callback: (Result<OWUserProfile>) -> Unit) {
-    callback(getAuthenticatedUserProfileUseCase())
+  override fun authenticateUserPreferred(profileId: String, callback: (Result<OWRegistrationResponse>) -> Unit) {
+    authenticateUserUseCase(profileId, null, callback)
```

Comment:
Why do we not return the Result in this function as was done previously? This should make it more testable?

Reply:
Don't really like that null is being used as preferred OWAuthenticatorType; can't we extend this and use a specific type to authenticate indicate preferred authentication?

Reply:
I initially had the same idea and actually implemented that. The problem with that we actually use this type when getting the authenticators aswel and then preferred shows as a possible type of authenticator and that's not right. 
```
class OWAuthenticator {
  String id;
  String name;
  bool isRegistered;
  bool isPreferred;
  OWAuthenticatorType authenticatorType;
}
```
Does that make sense?